2007-08-20  Corentin Le Gall <clegall@proformatique.com>

	pf_voicemail_date_language.patch
	$Revision$
	$Date$
	Bugfix

        The date language is appropriately set in the main body of
        voicemail-issued e-mails.

Index: asterisk-1.2.27/apps/app_voicemail.c
===================================================================
--- asterisk-1.2.27.orig/apps/app_voicemail.c	2008-03-19 11:33:16.000000000 +0100
+++ asterisk-1.2.27/apps/app_voicemail.c	2008-03-19 11:34:08.000000000 +0100
@@ -49,6 +49,7 @@
 #include <sys/mman.h>
 #include <time.h>
 #include <dirent.h>
+#include <locale.h>
 
 #include "asterisk.h"
 
@@ -1687,6 +1688,7 @@
 	struct vm_zone *the_zone = NULL;
 	int len_passdata;
 	char *passdata2;
+	locale_t loc;
 
 	if (vmu && ast_strlen_zero(vmu->email)) {
 		ast_log(LOG_WARNING, "E-mail address missing for mailbox [%s].  E-mail will not be sent.\n", vmu->mailbox);
@@ -1737,7 +1739,8 @@
 		fprintf(p, "Date: %s\n", date);
 
 		/* Set date format for voicemail mail */
-		strftime(date, sizeof(date), emaildateformat, &tm);
+		loc = newlocale (LC_ALL_MASK, "", NULL);
+		strftime_l(date, sizeof(date), emaildateformat, &tm, loc);
 
 		if (*fromstring) {
 			struct ast_channel *ast = ast_channel_alloc(0);
